src/Sarif.Driver/OrderedFileSpecifier.cs

```diff
@@ -103,10 +103,18 @@ private IEnumerable<IEnumeratedArtifact> EnumeratedArtifacts()
             }
             else
             {
-                WriteFilesInDirectoryToChannel(directory, filesToProcessChannel, filter, new SortedSet<string>(StringComparer.Ordinal));
-
-                filesToProcessChannel.Writer.Complete();
-                directoryEnumerationTask = Task.CompletedTask;
+                directoryEnumerationTask = Task.Run(() =>
```

Comment:
This is the only fix needed to work, I tested run with it modified directly in SPMI.